### ISSUE TITLE
Request for Assistance with LLaST Model Implementation

### ISSUE THREAD

**Issue by Lalaramarya** [openaudiolab/LLaST#2]
[tmp_inference_results_e2_fr_en.json](https://github.com/user-attachments/files/18260324/tmp_inference_results_e2_fr_en.json)
Dear Authors,

I recently read your paper, "LLaST: Improved End-to-end Speech Translation System Leveraged by Large Language Models," and I want to thank you for sharing the training code. I have started implementing the code, and the training process began without any issues.

After training the model for one epoch, I observed that the evaluation results included repetitive words that had no meaning. To address this, I continued training for an additional three epochs, but unfortunately, the issue persisted without any improvement.

I am writing to kindly request access to the trained models if you can share them. Alternatively, could you please suggest strategies or modifications to help achieve correct and meaningful results? For your reference, I have attached the evaluation result.json file from my experiments.

**Comment by ChenX17** [comment]
Hello Lalaramarya,

Thank you for your interest in our work.

You can find the following line in the xtuner/tools/test.py file:
`runner.model.load_state_dict(state_dict, strict=False)`
Please check if the trained weights were successfully loaded.

**Comment by Lalaramarya** [comment]
@ChenX17 Thank you for your reply
The weights are successfully loaded, but still, during inference, it provides the results I shared in the file in my previous comment.
[
    {
        "predictions": "\ufffd\ufffd\ufffdn.\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n MSG:\n\n\n\n\n MSG:\n\n\n\n\n\n\n\n MSG:\n\n\n\n\n\n\n MSG:\n\n\n\n\n\n\n MSG:\n\n MSG:\n\n\n\n\n\n MSG:\n\n\n\n\n\n\n\n MSG:\n MSG:\n\n\n\n\n\n\n MSG:\n\n\n\n\n\n\n MSG:\n\n\n\n\n\n MSG:\n\n\n\n\n\n\n MSG:\n MSG:\n\n\n",
        "references": "So the family will need to relearn everything."
    },
    {
        "predictions": "\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd\ufffdheet\ufffd\ufffd",
        "references": "This site contains four tombstones from the Achaemenid dynasty and seven from the Sasanian dynasty."
    },
    {
        "predictions": "",
        "references": "I said that the actors of wood had, in my opinion, a lot of advantages over the others."
    },

**Comment by lormaechea** [comment]
Hi,

I am also facing the same problem. I have fine-tuned TinyLlama for both 1 and 10 epochs, and I get highly monotonous results in both scenarios.

The model weights are correctly loaded.

``` 
[
    {
        "predictions": "Please translate the French sentence into English.",
        "references": "So the family will need to relearn everything."
    },
    {
        "predictions": "<|assistant|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:\nSure|:",
        "references": "This site contains four tombstones from the Achaemenid dynasty and seven from the Sasanian dynasty."
    },
    {
        "predictions": "<|assistant|>\nPlease translate the French sentence into English.",
        "references": "I said that the actors of wood had, in my opinion, a lot of advantages over the others."
    },
    {
        "predictions": "<|assistant|:\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into English.\nPlease translate the French sentence into",
        "references": "However, harmonization is in progress, at least on indirect taxes."
    },
``` 

Is there a way to fix this behavior?

Thanks in advance.

Best.

**Comment by ChenX17** [comment]
@lormaechea @Lalaramarya Could you provide some information about the loss during training?
